### PR TITLE
feat(fn): added fn-list component and generic approaches

### PR DESCRIPTION
**Pull request by g-cheishvili**
## Related Issue(s)

Part of #7670 

## Description

Implemented Fiori next list component. Intentionally did not wrote unit tests and also there is mixin extension in ListItemComponent class, that will need to be replaced with recommended approach, once we recommend how to deal with such scenario.

### After:
image

### CODE REVIEW

**Review by mikerodonnell89 — CHANGES_REQUESTED** [review]
The readonly list items aren't working, should render as a grey list item with no hover interactions https://fundamental-styles.netlify.app/?path=/docs/btp-fn-components-list--byline&globals=fioriVersion:fioriNext#list

**Comment by mikerodonnell89** [comment]
I really like the fnDisabled directive. In a separate PR, we should add a "Utilities" section to the side nav, documentation on how to use the disabled/readonly directives, which demonstrates buttons to turn on and off the disabling like this

```
<ul fn-list>
    <li fn-list-item [fnDisabled]="isInfoBarDisabled" infoBar>
        <span *fnListItemTitle> Disabled info bar </span>
    </li>
</ul>

<button fd-button (click)="isInfoBarDisabled = !isInfoBarDisabled">Toggle info bar disabled</button>
```

and some examples on other components such as button etc.

File: libs/fn/src/lib/cdk/mixins/CombineBehaviors.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+// import { Constructor } from '../interfaces/Constructor';
```
Comment (InnaAtanasova) [inline libs/fn/src/lib/cdk/mixins/CombineBehaviors.ts:1]: if this file is not used maybe remove it?

**Comment by InnaAtanasova** [comment]
Screen Shot 2022-02-18 at 12 39 33 PM

disabled item is focussable

**Comment by g-cheishvili** [comment]
> 

That happens on netlify deploy, it not up to date. look [here](https://fundamental-ngx-gh--pr7767-feat-fn-list-kbhl2j6r.web.app/fundamental-ngx#/fn/list)

**Comment by mikerodonnell89** [comment]
> > The readonly list items aren't working, should render as a grey list item with no hover interactions https://fundamental-styles.netlify.app/?path=/docs/btp-fn-components-list--byline&globals=fioriVersion:fioriNext#list
> 
> seems like only infoBar has readonly support. is that how it should be or styles is missing something?

You are correct, my bad, but let's make this line an infobar as well so that the fnReadonly directive actually does something

**Comment by g-cheishvili** [comment]
Go away stale bot

**Review by mikerodonnell89 — APPROVED** [review]

**Pull request merged**